### ISSUE TITLE
Proseg-to-baysor ParseError("Error while parsing value transcript_id for column 0 at line 0")

### ISSUE THREAD

**Issue by devianggr** [dcjones/proseg#31]
Hi,

I have an issue converting the CSV and geojson files to baysor-compatible files for Xenium Ranger running.
The issue that I received, was like this:
![Parse error-proseg to baysor](https://github.com/user-attachments/assets/601604ac-53fb-4366-a837-677dcef2f8d8)

I don't know what the problem is in this case. But I hope you could help me solving this issue soon.

Thanks a lot!

Devi

**Comment by brandenjlynch** [comment]
Seeing the same issue -- the format of my transcripts file, after running proseg, looks like this:
`transcript_id,x,y,z,observed_x,observed_y,observed_z,gene,qv,fov,assignment,probability,background,confusion
281500746519737,220.87396,17372.799,16.035217,221.47133,17372.148,13.866151,Hp,11.953445,AA3,4294967295,1.00001,1,0`

**Comment by asmilagswash** [comment]
Seeing the same issue , i would just like to add i dont see any cell column in the transcript-metdata.csv but I think the script needs a "cell" column

**Comment by asmilagswash** [comment]
I just changed the cell id column to "assignment" and it worked for me

**Comment by asmilagswash** [comment]
# This is how I modified the script  

#!/usr/bin/env python3
import argparse
import pandas as pd
import json
import os
import gzip

BACKGROUND_CELL = 2**32 - 1

def read_proseg_transcript_metadata(filename):
    # Determine file format
    if filename.endswith(".csv"):
        df = pd.read_csv(filename)
    elif filename.endswith(".csv.gz"):
        df = pd.read_csv(filename, compression='gzip')
    elif filename.endswith(".parquet"):
        df = pd.read_parquet(filename)
    else:
        raise ValueError(f"Unsupported file format for {filename}")

    # Check for required columns
    required_columns = ['transcript_id', 'assignment', 'x', 'y', 'z', 'qv']
    missing_columns = [col for col in required_columns if col not in df.columns]
    if missing_columns:
        raise ValueError(f"Missing columns in input file: {', '.join(missing_columns)}")

    return df

def cell_populations(metadata_df, min_qv):
    if 'assignment' not in metadata_df.columns:
        raise KeyError("'assignment' column not found in the metadata DataFrame")

    # Calculate cell populations
    cell_population = metadata_df['assignment'].apply(lambda x: 0 if x == BACKGROUND_CELL else 1).astype(int)
    return cell_population.groupby(metadata_df['assignment']).size().to_dict()

def read_cell_polygons_geojson(filename):
    if filename.endswith(".gz"):
        # Handle gzipped files
        with gzip.open(filename, 'rt', encoding='utf-8') as file:
            data = json.load(file)
    else:
        # Handle plain JSON files
        with open(filename, 'r', encoding='utf-8') as file:
            data = json.load(file)

    if 'features' not in data:
        raise KeyError("'features' key not found in GeoJSON data")

    features = data['features']
    geometries = []

    for feature in features:
        geometry = feature.get('geometry', {})
        coordinates = geometry.get('coordinates', [])
        if not coordinates:
            continue
        
        largest_polygon = max(coordinates, key=lambda poly: len(poly))  # Replace with actual area calculation if needed
        geometries.append({
            'type': 'Polygon',
            'coordinates': largest_polygon,
            'properties': feature.get('properties', {})
        })

    return data, geometries

def write_baysor_transcript_metadata(filename, metadata_df):
    metadata_df.to_csv(filename, index=False)

def write_cell_polygon_geojson(filename, data, geometries):
    data['geometries'] = geometries
    data['type'] = 'GeometryCollection'

    with open(filename, 'w') as file:
        json.dump(data, file)

def main():
    parser = argparse.ArgumentParser(description="Convert Proseg output to Baysor-compatible format.")
    parser.add_argument('transcript_metadata', type=str, help='Path to transcript metadata file')
    parser.add_argument('cell_polygons', type=str, help='Path to cell polygons GeoJSON file')
    parser.add_argument('--output_transcript_metadata', type=str, default='baysor_transcript_metadata.csv', help='Path to output transcript metadata file')
    parser.add_argument('--output_cell_polygons', type=str, default='baysor_cell_polygons.geojson', help='Path to output cell polygons GeoJSON file')
    args = parser.parse_args()

    # Read input files
    metadata_df = read_proseg_transcript_metadata(args.transcript_metadata)
    _, geometries = read_cell_polygons_geojson(args.cell_polygons)

    # Process data
    cell_population = cell_populations(metadata_df, 20.0)

    # Write outputs
    write_baysor_transcript_metadata(args.output_transcript_metadata, metadata_df)
    write_cell_polygon_geojson(args.output_cell_polygons, {}, geometries)  # 'data' needs to be replaced with actual data if necessary

if __name__ == "__main__":
    main()

**Comment by came203** [comment]
Seeing the same issue here, my transcript-metadata.csv.gz looks like:
```

transcript_id,x,y,z,observed_x,observed_y,observed_z,gene,qv,fov,assignment,probability,background,confusion
0,11133.598,7328.0205,2.3660069,11133.936,7327.5967,0.0,Ngfr,inf,0,17056,0.21,0,1
```

Command: 
`proseg-to-baysor transcript-metadata.csv.gz cell-polygons.geojson.gz --output-transcript-metadata baysor-transcript-metadata.csv --output-cell-polygons baysor-cell-polygons.geojson`

**Comment by dcjones** [comment]
Apologies for the regression introduced here. It's now fixed in version 1.1.4. It should not be necessary to re-run proseg.

**Comment by brandenjlynch** [comment]
Thank you! I will give it a try now.
